src/posts/lucky_1.0.0-rc1.cr

```diff
@@ -0,0 +1,239 @@
+
+    ### How to upgrade
+
+    See the [UPGRADE_NOTES](https://github.com/luckyframework/lucky/blob/main/UPGRADE_NOTES.md#upgrading-from-029-to-030).
+
+    You can also view the changes using [LuckyDiff](https://luckydiff.com/?from=0.30.0&to=1.0.0-rc1).
+    This tool helps see what has changed between versions and will be included in upgrade notes from now on.
+
+    ## Here's what's new
+
+    This version releases several breaking changes that were needed in
+    order to make the framework more extensible for the future.
+
+    ### No more DB dependency
+
+    Many sites (including this one) have no need for a databaase. In some cases,
+    your data comes from a 3rd party API, or maybe you need a custom database
+    engine other than PostgreSQL.
+
+    In this release, Avram is no longer a dependency of Lucky, but instead,
+    Lucky is now an optional dependency of Avram. If you're using Avram outside
+    of Lucky, then nothing changes for you. However, for Lucky users using Avram,
+    you'll now have to include it as an additional dependency. Be sure to add it
+    to your `shard.yml` file, and require it in your `src/shards.cr`.
+
+    ### Support for Array params in operations
+
+    Support for Arrays in Lucky and Avram have existed for a few versions now.
+    This includes support for arrays in params. However, one bit that has been
+    missing was the ability to pass those array params through to an operation.
+
+    For example:
+
+    ```crystal
+    # before this release
+    # post:tags[]=lucky&post:tags[]=crystal
+    tags = params.nested_arrays("post")["tags"]
+    SavePost.create(params, tags: tags) do |o, p|
+      # ...
+    end
+
+    # after this release
+    SavePost.create(params) do |o, p|
+      # ...
+    end
+    ```
+
+    This ended up being quite a big breaking change. The original Params implementation
+    assumed the type was `Hash(String, String)`, but this meant you couldn't have an
+    array value. Now the type assumes `Hash(String, Array(String))` which is actually
+    the same type as [`URI::Params`](https://crystal-lang.org/api/1.5.1/URI/Params.html#new%28raw_params%3AHash%28String%2CArray%28String%29%29%29-class-method).
+
+    ### Changes to the javascript Ecosystem
+
+    As we all know, the javascript ecosystem changes in leaps and bounds daily.
+    This makes it very difficult to keep up.
+
+    In 2017, when Lucky began, the Rails core team had a neat library called
+    [Turbolinks](https://github.com/turbolinks/turbolinks-classic).
+    This library helped to make pages feel even faster, and who doesn't love speed?
+    Lucky included this library by default, but it has since been deprecated in favor
+    of [Turbo](https://turbo.hotwired.dev/). Although Lucky doesn't include Turbo currently,
+    we would still like to encourage users to give it a shot, and let us know if this
+    is something you'd like to see included by default.
```

Comment:
I am not sure from reading this what has changed. 

> but it has since been deprecated in favor of [Turbo]

Does that mean we have deprecated Turbolinks in favor of Turbo, or that Rails has and we have deprecated both? And when it says encourage users to give it a shot do you mean encourage people to add Turbo and see what they think?

If so then maybe something like this:

```suggestion
    [Turbolinks](https://github.com/turbolinks/turbolinks-classic).
    This library helped to make pages feel even faster, and who doesn't love speed?
    Lucky included Turbolinks by default, but Rails has since deprecated it in favor
    of [Turbo](https://turbo.hotwired.dev/). Lucky now doesn't include either library, but
    we would still like to encourage users to give Turbo a shot, and let us know if it
    is something you'd like to see included by default.
```

I'd also be curous to hear why we removed Turbolinks without replacing it with Turbo. I'm not saying it was bad/good just curious why. I haven't been following it too closely so am not sure of the pros/cons and I think readers may also be curious about why we removed it

Reply:
We removed turbolinks because the official repo is now archived by Rails. I didn't add in Turbo yet because there's a few questions I have unanswered like, do we need to do redirect stuff like Turbolinks had? Also, there's a few edge cases with Turbo where you need use a `<turbo-frame></turbo-frame>` tag (e.g. when an action redirects to itself), but what are the other cases? I just don't know enough about Turbo to add it confidently.

Reply:
I guess I should also say that I'd be cool with adding Turbo in as a default, which is why I want people to try it out and see what needs to happen for that

Reply:
That makes total sense to me! Can't really add it if we can't be confident we've got it all wired up just right. Turbo can be tricky to test all the edge cases